File: AndroidSerializable/Parcelable/app/src/main/java/com/haoyudu/parcelable/Student.java
package com.haoyudu.parcelable;

import android.os.Parcel;
import android.os.Parcelable;

import java.io.Serializable;

/**实现android自带的Parcelable接口*/
public class Student implements Parcelable {
    private static final long serialVersionUID = 3369452620490726091L;
    private String name;
    private int age;
    private Score score;

    public Student(String name,int age,Score score){
        this.name=name;
        this.age=age;
        this.score=score;
    }

    protected Student(Parcel in) {
        name = in.readString();
        age = in.readInt();
        score = in.readParcelable(Score.class.getClassLoader());
    }

    public static final Creator<Student> CREATOR = new Creator<Student>() {
        @Override
        public Student createFromParcel(Parcel in) {
            return new Student(in);
        }

        @Override
        public Student[] newArray(int size) {
            return new Student[size];
        }
    };

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Score getScore() {
        return score;
    }

    public void setScore(Score score) {
        this.score = score;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(name);
        dest.writeInt(age);
        dest.writeParcelable(score, flags);
    }
}

File: AndroidDatabase/RoomDatabaseSimpleOperate/RoomCRUD/app/src/main/java/com/haoyudu/roomcrud/entity/Student.kt
package com.haoyudu.roomcrud.entity

import android.os.Parcel
import android.os.Parcelable
import androidx.room.Entity

@Entity
data class Student(var name: String?, var classmate: String?, var age:Int) : Parcelable {

    var id:Int=0

    constructor():this("user","null",0)

    constructor(parcel: Parcel) : this(
        parcel.readString(),
        parcel.readString(),
        parcel.readInt()
    ) {
        id = parcel.readInt()
    }

    override fun describeContents(): Int {
        TODO("Not yet implemented")
    }

    override fun writeToParcel(dest: Parcel?, flags: Int) {
        TODO("Not yet implemented")
    }

    companion object CREATOR : Parcelable.Creator<Student> {
        override fun createFromParcel(parcel: Parcel): Student {
            return Student(parcel)
        }

        override fun newArray(size: Int): Array<Student?> {
            return arrayOfNulls(size)
        }
    }

}